internal/controller/storageclient_controller.go

```diff
@@ -210,7 +210,21 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 			if err := r.reconcileClusterResourceQuota(clusterResourceQuotaSpec); err != nil {
 				return reconcile.Result{}, err
 			}
+		case "ConfigMap":
+			data := map[string]string{}
+			err := json.Unmarshal(eResource.Data, &data)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to unmarshal ConfigMap data: %v", err)
+			}
+			if eResource.Name == "DesiredClientConfig" {
+				r.storageClient.Status.DesiredClientConfigHash = utils.GetMD5Hash(fmt.Sprintf("%v", data))
+				utils.AddAnnotation(r.storageClient, utils.DesiredSubscriptionChannelAnnotationKey, data["DesiredClientOperatorChannel"])
+				if err := r.update(r.storageClient); err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to update StorageClient with DesiredClientOperatorChannel annotation: %v", err)
+				}
+			}
```

Comment:
thinking about it, we don't need a specialized structure for the hash. Provider has a state spread across multiple resources, you consolidate that state and send that as hearbeat response which triggers a reconcile on storageclient.

During the reconcile, it gets the info that it requires via usual resources, be it configmap, quota, new csi-op APIs etc. We are only trying to invoke the controller when it is required to reconcile and hash is the differentiating factor.

Reply:
okay, I'm removing it in the updated commit.